src/main/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtil.java

```diff
@@ -135,11 +213,56 @@ private static boolean addControlledFieldToMarcRecord(Record record, String fiel
         }
       }
     } catch (Exception e) {
-      LOGGER.warn("addControlledFieldToMarcRecord:: Failed to add additional controlled field {} to record {}", field, record.getId(), e);
+      LOGGER.warn("addControlledFieldToMarcRecord:: Failed to add additional controlled field {} to record {}",
+        field, getRecordId(record), e);
     }
     return result;
   }
 
+  public static void addControlledFieldToMarcRecord(String field, String value, org.marc4j.marc.Record marcRecord) {
+    ControlField dataField = MarcFactory.newInstance().newControlField(field, value);
+    marcRecord.addVariableField(dataField);
+  }
+
+  public static void replaceControlledFieldInMarcRecord(String field, String value, org.marc4j.marc.Record marcRecord) {
+    ControlField currentField = (ControlField) marcRecord.getVariableField(field);
+    if (currentField != null) {
+      ControlField newControlField = MarcFactory.newInstance().newControlField(field, value);
+      marcRecord.getControlFields().set(marcRecord.getControlFields().indexOf(currentField), newControlField);
+    }
+  }
+
+  /**
+   * Move original marc hrId to 035 tag
+   *
+   * @param record input record to modify
+   */
+  public static void fillHrIdFieldInMarcRecord(Record record) {
+    String hrId = getValueFromControlledField(record, HR_ID_PREFIX_FROM_FIELD);
+    String valueFrom001 = getValueFromControlledField(record, HR_ID_FROM_FIELD);
+    if (!StringUtils.equals(hrId, valueFrom001)) {
```

Comment:
Is this comparison needed for values from the 001 and 003 fields? Just asking because it looks like such a comparison does not exist in the current implementation in the record-storage module.